doc/api/repl.md

```diff
@@ -909,30 +909,226 @@ to connect to both Unix and TCP sockets.
+
+process.stdin.pipe(sock);
+sock.pipe(process.stdout);
+
+sock.on('connect', () => {
+  process.stdin.resume();
+  process.stdin.setRawMode(true);
+});
+
+sock.on('close', () => {
+  process.stdin.setRawMode(false);
+  process.stdin.pause();
+  sock.removeListener('close', done);
+});
+
+process.stdin.on('end', () => {
+  sock.destroy();
+  console.log();
+});
+
+process.stdin.on('data', (b) => {
+  if (b.length === 1 && b[0] === 4) {
+    process.stdin.emit('end');
+  }
+});
+```
+
+```cjs
+// repl-client.js
+const net = require('node:net');
+
+const sock = net.connect(1337);
+
+process.stdin.pipe(sock);
+sock.pipe(process.stdout);
+
+sock.on('connect', () => {
+  process.stdin.resume();
+  process.stdin.setRawMode(true);
+});
+
+sock.on('close', () => {
+  process.stdin.setRawMode(false);
+  process.stdin.pause();
+  sock.removeListener('close', done);
+});
+
+process.stdin.on('end', () => {
+  sock.destroy();
+  console.log();
+});
+
+process.stdin.on('data', (b) => {
+  if (b.length === 1 && b[0] === 4) {
+    process.stdin.emit('end');
+  }
+});
+```
+
+To run the example open two different terminals on your machine, start the server
+with `node repl-server.js` in one terminal and `node repl-client.js` on the other.
+
+Original code from <https://gist.github.com/TooTallNate/2209310>.
+
+#### REPL over `curl`
+
+This is an example on how to run a REPL instance over [`curl()`][]
+
+The following script starts an HTTP server on port `8000` that can accept
+a connection established via [`curl()`][].
+
+```mjs
+import http from 'node:http';
+import repl from 'node:repl';
+import { Buffer } from 'node:buffer';
+
+const buf0 = Buffer.from([0]);
+
+const server = http.createServer((req, res) => {
+  res.setHeader('content-type', 'multipart/octet-stream');
+
+  repl.start({
+    prompt: 'curl repl> ',
+    input: req,
+    output: res,
+    terminal: false,
+    useColors: true,
+    useGlobal: false,
+  });
+
+  // Hack to thread stdin and stdout
+  // simultaneously in curl's single thread
+  const iv = setInterval(() => {
+    res.write(buf0);
+  }, 100);
+
+  req.on('close', () => {
+    clearInterval(iv);
+  });
+});
+server.listen(8000);
+```
+
+```cjs
+const http = require('node:http');
+const repl = require('node:repl');
+const buf0 = Buffer.from([0]);
+
+const server = http.createServer((req, res) => {
+  res.setHeader('content-type', 'multipart/octet-stream');
+
+  repl.start({
+    prompt: 'curl repl> ',
+    input: req,
+    output: res,
+    terminal: false,
+    useColors: true,
+    useGlobal: false,
+  });
+
+  // Hack to thread stdin and stdout
+  // simultaneously in curl's single thread
+  const iv = setInterval(() => {
+    res.write(buf0);
+  }, 100);
+
+  req.on('close', () => {
+    clearInterval(iv);
+  });
+});
+server.listen(8000);
+```
+
+When the above script is running you can then use [`curl()`][] to connect to
+the server and connect to its REPL instance by running `curl -sSNT. localhost:8000`
```

Comment:
```suggestion
the server and connect to its REPL instance by running `curl -sSNT. localhost:8000`.
```

Reply:
I also need to add `--no-progress-meter` flag, otherwise the output is broken
```suggestion
the server and connect to its REPL instance by running `curl --no-progress-meter -sSNT. localhost:8000`.
```

Reply:
good call, yeah I thought that that was sort of expected :confused:  (and quite annoying :sweat_smile:)